Add swagger gradle file to generate java client api

adazza-daniel opened the pull request:
@ben-adazza

adazza-daniel committed:
Add swagger gradle file to generate java client api

adazza-daniel committed:
Minor cleanup: Remove unneeded library dependencies, etc.

ben-adazza commented on adazza-swagger.gradle:
```diff
@@ -0,0 +1,68 @@
+dependencies {
+    swaggerCompile "org.springframework.boot:spring-boot-starter-web:1.5.6.RELEASE"
+    swaggerCompile 'io.swagger:swagger-annotations:1.5.16'
+    swaggerCompile 'com.squareup.okhttp:okhttp:2.7.5'
```
Why are these provided here and in runtime scope?

ben-adazza replied:
Also these versions need to move into the dep mgmt file.

ben-adazza commented on adazza-swagger.gradle:
```diff
@@ -0,0 +1,68 @@
+
+    additionalProperties = [
+            'invokerPackage': 'io.swagger.kylo.client',
+            'modelPackage'  : 'io.swagger.kylo.client.model',
```
You need to figure out a way to have this configured in the consuming cod.e

adazza-daniel replied:
When you say consuming code do you mean the consuming gradle file that imports this?

ben-adazza replied:
yes.

ben-adazza commented on adazza-swagger.gradle:
```diff
@@ -0,0 +1,68 @@
+    swaggerCompile 'com.squareup.okhttp:logging-interceptor:2.7.5'
+    swaggerCompile 'com.google.code.gson:gson:2.8.1'
+
+    compile sourceSets.swagger.output
```
This should happen automatically.

adazza-daniel committed:
Simplify/refactor adazza-swagger.gradle, move swagger dependencies in adazza dep management.

ben-adazza commented on adazza-dependency-management.gradle:
```diff
@@ -44,6 +44,13 @@ allprojects {
 
 
+      // Swagger: swagger-codegen java client API dependencies
+      dependency group: "org.springframework.boot", name: "spring-boot-starter-web", version: "1.5.6.RELEASE"
```
Move all these  version to dep-mgmt.

ben-adazza commented on adazza-swagger.gradle:
```diff
@@ -0,0 +1,36 @@
+apply plugin: org.detoeuf.SwaggerCodeGenPlugin
+
+configurations {
+    swagger
```
What does this accomplish?

adazza-daniel commented:
@ben-adazza updated the PR addressing your comments. Many of lines of code in which you commented on are now removed, moved lib deps in dep-mgmt, reduced lib deps.

ben-adazza commented on adazza-swagger.gradle:
```diff
@@ -0,0 +1,32 @@
+sourceSets {
+    main {
+        java {
+            srcDir file("${project.buildDir.path}/swagger/src/main/java")
```
for this to look like scrooge it would be:
`buildDir/generated-sources/` the src/main/java is superfluous.
